jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcDataContext.java

```diff
@@ -231,7 +232,27 @@ private JdbcDataContext(DataSource dataSource, Connection connection, TableType[
         } else if (DATABASE_PRODUCT_H2.equals(_databaseProductName)) {
             setQueryRewriter(new H2QueryRewriter(this));
         } else if (DATABASE_PRODUCT_HIVE.equals(_databaseProductName)) {
-            setQueryRewriter(new HiveQueryRewriter(this));
+            int majorVersion = 0;
+            String[] parts = databaseVersion.split("\\.");
+            if(parts.length < 2) {
+                logger.warn("Illegal Hive Version: " + databaseVersion + " (expected A.B.* format)");
+                parts[0] = "0";
+            } else {
+                majorVersion = Integer.valueOf(parts[0]);
+            }
+            switch (majorVersion){
+                case 1:
```

Comment:
I suggest moving this down before the `default` case, then you don't need to repeat the case's code block